plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -227,4 +239,30 @@ private static String quoteIdentifier(String identifier)
     {
         return format("\"%s\"", identifier.replaceAll("\"", "\"\""));
     }
+
+    public static Optional<String> getQueryOptionsString(String options)
+    {
+        if (isNullOrEmpty(options)) {
+            return Optional.empty();
+        }
+
+        Map<String, String> queryOptionsMap = ImmutableMap.copyOf(MAP_SPLITTER.split(options));
+        return getQueryOptions(queryOptionsMap);
+    }
+
+    public static Optional<String> getQueryOptions(Map<String, String> queryOptionsMap)
+    {
+        if (queryOptionsMap.isEmpty()) {
+            return Optional.empty();
+        }
+        StringBuilder result = new StringBuilder();
+        for (Map.Entry<String, String> entry : queryOptionsMap.entrySet()) {
+            result.append("SET ")
+                    .append(entry.getKey())
+                    .append(" = ")
+                    .append(entry.getValue())
+                    .append(";\n");
+        }
+        return Optional.of(result.toString());
+    }
```

Comment:
Use https://github.com/trinodb/trino/blob/df454dbdb6b02ca0429ee341f844add139c3e1c5/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHttpMetastoreConfig.java#L83-L113 as PoC on how to deal with a config value containing a map of values.

Reply:
@naman-patel I was suggesting returning a `Map<String, String>`

Reply:
But then everywhere there needs to be code to translate the settings to SET statements. That feels like redundant code everywhere.

Reply:
This config getter is called from `PinotSessionProperties` and it expects String type. Returning `Optional<String>` seems acceptable. We should still validate if the value can be converted to map though.